### DISCUSSION TITLE
Bucket exists, but Supabase says it doesn't

### DISCUSSION THREAD

**Discussion by GreatTA1998** [supabase/supabase#1761] [Questions] [answered]
I was able to successfully upload and download a photo in my Svelte app in the bucket with id "photos". However, when I call the operations `getBucket()`, it gives me an error 400 "Not found", and when I call `listBuckets()`, it returns an empty array.

I removed all the permissions policies, but the bug remains. Any help appreciated!
![image](https://user-images.githubusercontent.com/26662928/120083953-59432000-c09a-11eb-9fe8-055a2f83e14d.png)

**Comment by GreatTA1998** [comment, marked as answer]
Actually by removing all my storage bucket policies, I'm not able to even download the photo now. That makes me think that this bug is to do with permissions - that I had sufficient permissions to upload and download photos, but insufficient permissions to access the bucket / list buckets. But how do I alter the permissions to allow those things (without user authentication)?

**Reply by inian** [reply]
Hi @eltonlin1998, the new storage policy editor lets you add policies only to the objects table. For enabling users to [list buckets](https://supabase.io/docs/reference/javascript/storage-listbuckets), you need to add select permissions to the buckets table. To allow listing all buckets everyone, can you add the following policy in the SQL editor?
```sql
create policy list_all_buckets
on storage.buckets for select using (
true
);
```
Instead of `true` above, you can restrict buckets and users here too.

**Reply by GreatTA1998** [reply]
Hi Inian - thanks a lot for the help - it fixed the issue wonderfully. I'm not very familiar with SQL (I was a Firebase user), but now I have discovered a new world and will try to get myself familiar : ) thanks again.
